src/AddTodoForm.js

```diff
@@ -1,18 +1,29 @@
 import React from 'react';
 
-export default function AddTodoForm(props) {
+export default function AddTodoForm({ onAddTodo }) {
+  const [todoTitle, setTodoTitle] = React.useState('');
+
+  function handleTitleChange(event) {
+    const newTodoTitle = event.target.value;
+    setTodoTitle(newTodoTitle);
+  }
+
   function handleAddTodo(event) {
     event.preventDefault();
-    const todoTitle = event.target.title.value;
-    console.log(todoTitle);
-    props.onAddTodo(todoTitle);
-    event.target.reset();
+    onAddTodo({ title: todoTitle, id: Date.now() });
+    setTodoTitle('');
   }
 
   return (
     <form onSubmit={handleAddTodo}>
       <label htmlFor="todoTitle">Title</label>
-      <input type="text" id="todoTitle" name="title" />
+      <input
+        type="text"
+        id="todoTitle"
+        name="title"
+        value={todoTitle}
+        onChange={handleTitleChange}
+      />
```

Comment:
I don't remember if I've made this suggestion before (or if it'll come up in the next lesson or something) but it's a good idea to add the `required` attribute to inputs so that when you click the button with no text entered it doesn't create a new blank todo. 

```suggestion
      <input
        type="text"
        id="todoTitle"
        name="title"
        value={todoTitle}
        onChange={handleTitleChange}
        required
      />
```